forward-proxy/src/statistics/mod.rs

```diff
@@ -0,0 +1,132 @@
+use crate::config::InfluxDBConfig;
+use futures::stream;
+use influxdb2::Client;
+use influxdb2::models::DataPoint;
+use pingora::http::StatusCode;
+use pingora::proxy::Session;
+use pingora_router::ctx::{Layer8Context, Layer8ContextTrait};
+use std::error::Error;
+
+struct InfluxDBMeasurements;
+
+impl InfluxDBMeasurements {
+    const TOTAL_BYTE_TRANSFERRED: &'static str = "total_byte_transferred";
+    const TOTAL_TUNNEL_INITIATED: &'static str = "total_tunnel_initiated";
+    const TOTAL_SUCCESS: &'static str = "total_success";
+    const TOTAL_REQUEST: &'static str = "total_request";
+}
+
+pub struct InfluxDBClient {
+    client: Client,
+    bucket: String,
+}
+
+impl InfluxDBClient {
+    pub fn new(config: &InfluxDBConfig) -> Self {
+        let influxdb_client = Client::new(
+            &config.influxdb_url,
+            &config.influxdb_org,
+            &config.influxdb_auth_token,
+        );
+        InfluxDBClient {
+            client: influxdb_client,
+            bucket: config.influxdb_bucket.clone(),
+        }
+    }
+
+    pub async fn update_statistics(
+        &self,
+        client_id: &str,
+        session: &Session,
+        ctx: &Layer8Context,
+        response_status: u16,
+    ) -> Result<(), Box<dyn Error + Sync + Send>> {
+        self.increase_total_request(client_id).await?;
```

Comment:
Here true async code would be better. Blocking with `await` could probably be avoided with a clever solution that, if an error occurs, we handle it gracefully. The 'cost' might be that we miss out on the monetization of an incremented requests but the benefit might be more performant code.